orchagent/bfdorch.cpp

```diff
@@ -551,3 +615,116 @@ uint32_t BfdOrch::bfd_src_port(void)
     return (port++);
 }
 
+void BfdOrch::notify_session_state_down(const string& key)
+{
+    size_t found_vrf = key.find(delimiter);
+    if (found_vrf == string::npos)
+    {
+        SWSS_LOG_ERROR("Failed to parse key %s, no vrf is given", key.c_str());
+        return;
+    }
+
+    size_t found_ifname = key.find(delimiter, found_vrf + 1);
+    if (found_ifname == string::npos)
+    {
+        SWSS_LOG_ERROR("Failed to parse key %s, no ifname is given", key.c_str());
+        return;
+    }
+    string vrf_name = key.substr(0, found_vrf);
+    string alias = key.substr(found_vrf + 1, found_ifname - found_vrf - 1);
+    IpAddress peer_address(key.substr(found_ifname + 1));
+    BfdUpdate update;
+    update.peer = get_state_db_key(vrf_name, alias, peer_address);
+    update.state = SAI_BFD_SESSION_STATE_DOWN;
+    notify(SUBJECT_TYPE_BFD_SESSION_STATE_CHANGE, static_cast<void *>(&update));
+    return;
+}
+
+void BfdOrch::handleTsaStateChange(bool tsaState)
+{
+    SWSS_LOG_INFO("BfdOrch TSA state Changed to %d.\n", int(tsaState));
+    tsa_enabled = tsaState;
+    for (auto it : bfd_session_cache)
+    {
+        if (tsaState == true)
+        {
+            notify_session_state_down(it.first);
+            if (!remove_bfd_session(it.first))
+            {
+                SWSS_LOG_ERROR("Failed to remove BFD session %s\n", it.first.c_str());
```

Comment:
Please remove the error log as its handled in the remove function already

Reply:
done